workitem/link/link_repository_blackbox_test.go

```diff
@@ -40,296 +26,169 @@ func TestRunLinkRepoBlackBoxTest(t *testing.T) {
-	require.Nil(s.T(), err)
-	s.child, err = s.createWorkitem(workitem.SystemBug, "Child", workitem.SystemStateNew)
-	require.Nil(s.T(), err)
-}
-
-func (s *linkRepoBlackBoxTest) createWorkitem(wiType uuid.UUID, title, state string) (*workitem.WorkItem, error) {
-	return s.workitemRepo.Create(
-		s.Ctx, s.testSpace, wiType,
-		map[string]interface{}{
-			workitem.SystemTitle: title,
-			workitem.SystemState: state,
-		}, s.testIdentity.ID)
 }
 
-// This creates a parent-child link between two workitems -> parent1 and Child. It tests that when there is an attempt to create another parent (parent2) of child, it should throw an error.
-func (s *linkRepoBlackBoxTest) TestDisallowMultipleParents() {
-	// create a work item link
-	_, err := s.workitemLinkRepo.Create(s.Ctx, s.parent1.ID, s.child.ID, s.testTreeLinkTypeID, s.testIdentity.ID)
-	require.Nil(s.T(), err)
-	// when
-	_, err = s.workitemLinkRepo.Create(s.Ctx, s.parent2.ID, s.child.ID, s.testTreeLinkTypeID, s.testIdentity.ID)
-	// then
-	require.NotNil(s.T(), err)
-}
-
-// TestCountChildWorkitems tests total number of workitem children returned by list is equal to the total number of workitem children created
-// and total number of workitem children in a page are equal to the "limit" specified
-func (s *linkRepoBlackBoxTest) TestCountChildWorkitems() {
-	// create 3 workitems for linking as children to parent workitem
-	child1, err := s.createWorkitem(workitem.SystemBug, "Child 1", workitem.SystemStateNew)
-	require.Nil(s.T(), err)
-	child2, err := s.createWorkitem(workitem.SystemBug, "Child 2", workitem.SystemStateNew)
-	require.Nil(s.T(), err)
-	child3, err := s.createWorkitem(workitem.SystemBug, "Child 3", workitem.SystemStateNew)
-	require.Nil(s.T(), err)
-
-	// link the children workitems to parent
-	_, err = s.workitemLinkRepo.Create(s.Ctx, s.parent1.ID, child1.ID, s.testTreeLinkTypeID, s.testIdentity.ID)
-	require.Nil(s.T(), err)
-
-	_, err = s.workitemLinkRepo.Create(s.Ctx, s.parent1.ID, child2.ID, s.testTreeLinkTypeID, s.testIdentity.ID)
-	require.Nil(s.T(), err)
-
-	_, err = s.workitemLinkRepo.Create(s.Ctx, s.parent1.ID, child3.ID, s.testTreeLinkTypeID, s.testIdentity.ID)
-	require.Nil(s.T(), err)
-
-	offset := 0
-	limit := 1
-	res, count, err := s.workitemLinkRepo.ListWorkItemChildren(s.Ctx, s.parent1.ID, &offset, &limit)
-	require.Nil(s.T(), err)
-	require.Len(s.T(), res, 1)
-	require.Equal(s.T(), 3, int(count))
+func (s *linkRepoBlackBoxTest) TestList() {
+	// tests total number of workitem children returned by list is equal to the
+	// total number of workitem children created and total number of workitem
+	// children in a page are equal to the "limit" specified
+	s.T().Run("ok - count child work items", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, s.DB,
+			tf.WorkItems(4), // parent + child 1-3
+			tf.WorkItemLinkTypes(1, func(fxt *tf.TestFixture, idx int) error {
+				fxt.WorkItemLinkTypes[idx].ForwardName = "parent of"
```

Comment:
is there no constant for the `parent of` forward name of this link type ?

Reply:
not yet, but in space templates yes. 
@kwk right?

Reply:
Eventually this will hopefully all go away any time soon.